ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/Committer.scala

```diff
@@ -66,7 +66,7 @@ private[kvutils] trait Committer[Submission, PartialResult] {
       recordTime: Time.Timestamp,
       submission: Submission,
       participantId: ParticipantId,
-      inputState: DamlStateMap): (DamlLogEntry, Iterable[(DamlStateKey, DamlStateValue)]) =
+      inputState: DamlStateMap): (DamlLogEntry, Map[DamlStateKey, DamlStateValue]) =
```

Comment:
We have to be really careful here to retain ordering for the produced outputs (e.g. retain determinism). `Map` by defaults to `HashMap` which does not give a stable ordering. Use `TreeMap` instead, or retain the `Iterable` and produce a set of keys when verifying.

Reply:
As discussed offline, the consumer of this call will convert the result to a `Map` anyway. Let's review the data flows separately and ensure we have deterministic ordering where needed.